lib/workers/global/config/parse/file.ts

```diff
@@ -81,3 +91,26 @@ export async function getConfig(env: NodeJS.ProcessEnv): Promise<AllConfig> {
   }
   return config;
 }
+
+export async function deleteConfigFile(env: NodeJS.ProcessEnv): Promise<void> {
+  const configFile = env.RENOVATE_CONFIG_FILE;
+
+  if (is.undefined(configFile) || is.emptyStringOrWhitespace(configFile)) {
+    return;
+  }
+
+  if (env.RENOVATE_X_DELETE_CONFIG_FILE !== 'true') {
+    return;
+  }
```

Comment:
i see now a duplicate check, this second one should be removed this would be always false

Reply:
actually this was already pointed out by Rhys, 

https://github.com/renovatebot/renovate/pull/22857#discussion_r1233793150
> yes, it is intentional.
i dont want to invoke the function without any explicit safe checks, nor i want this function to delete the config without safe checks either as it is exported (for testing).
>
> we can maybe rename the function to tryDeleteConfigFile and remove the external safe checks.

i can remove it tho if needed :)

Reply:
Maybe remove the external check and always check inside the function